marlowe-playground-client/grammar.ne

```diff
@@ -3,7 +3,7 @@ const moo = require("moo");
 
 const lexer = moo.compile({
         WS: /[ \t]+/,
-        number: /0|-?[1-9][0-9]*/,
+        number: /-?[0-9](?:_*[0-9])*/,
```

Comment:
just to be sure, does anything test the parser? I think there's a test that generates random contracts but the generator hasn't been changed to generate `_`

Reply:
The thing is the generator creates a value, say `Constant 123` then converts it to String. It doesn't generate string "Constant 123". I could change a pretty printer to print `123000` as `123_000` but I'm not sure I should.

Reply:
And I added tests that test this.

Reply:
Ah ok, tricky.

Reply:
I fixed the parsers a bit, because Haskell allows literals to start with 0.
The parser is still slightly wrong, though. We sort of expect NegativeLiterals extension to be enabled in Haskell.
By default, Haskell parses `Constant -123` as `(-) Constant 123`.

Reply:
So we either need to enable `NegativeLiterals` or write `Constant (-123)`